boot/bootstate20.go

```diff
@@ -587,3 +690,65 @@ func (bsmark *bootState20MarkSuccessful) commit() error {
 
 	return nil
 }
+
+// genericEarlyBootChooseSnap will run the logic to choose which snap should be
+// mounted during the early boot, i.e. initramfs. Specifically, it only works
+// with kernel and base snaps. It returns the first and second choice for what
+// snaps to mount, if second is set, then it is the primary fallback snap, and
+// the first snap is the try snap, if second is unset, then first is the primary
+// fallback snap. It returns both so that a higher level function can do
+// additional verification of the try snap if it wants, such as the kernel snaps
+// being verified in the modeenv, or the base snaps updating the modeenv with
+// new values for base_status, etc.
+func genericEarlyBootChooseSnap(bs bootState, expectedTryStatus, typeString string) (
+	firstChoice, secondChoice snap.PlaceInfo,
+	err error,
+) {
+	curSnap, trySnap, snapTryStatus, err := bs.revisions()
+
+	if err != nil && !IsTrySnapError(err) {
+		// we have no fallback snap!
+		return nil, nil, fmt.Errorf("fallback %s snap unusable: %v", typeString, err)
+	}
+
+	// check that the current snap actually exists
+	file := curSnap.Filename()
+	snapPath := filepath.Join(dirs.SnapBlobDirUnder(filepath.Join(dirs.EarlyBootUbuntuData, "system-data")), file)
```

Comment:
this needs a rootdir concept?

Reply:
we resolved the larger issue here in the previous PR, now we can use boot pkg exported vars directly (i.e. `EarlyWritable`), then this code becomes essentially:

```suggestion
	snapPath := filepath.Join(dirs.SnapBlobDirUnder(EarlyWritable), file)
```

is that okay, or should we still pass in a rootdir somehow?